planner/core/exhaust_physical_plans.go

```diff
@@ -415,6 +414,32 @@ func (p *LogicalJoin) constructIndexMergeJoin(
 	return indexMergeJoins
 }
 
+func (p *LogicalJoin) constructIndexHashJoin(
+	prop *property.PhysicalProperty,
+	outerIdx int,
+	innerTask task,
+	ranges []*ranger.Range,
+	keyOff2IdxOff []int,
+	path *accessPath,
+	compareFilters *ColWithCmpFuncManager,
+) []PhysicalPlan {
+	// TODO(xuhuaiyu): support keep outer order for indexHashJoin.
```

Comment:
Do we need it? I thought in this case index merge join would always performs better.

Reply:
Some cases need to keep order but index merge join can't make it. For example:

We need to join tables `t1` and `t2` on `t1.c = t2.c`, `t1` is the outer table. But it requires to keep order with `t1.a, t1.b`. The index merge join can only make sure of keeping `t1.c` order.

Reply:
That's the reason why we also need index lookup join now.